meadow-client/src/Marlowe/Blockly.purs

```diff
@@ -0,0 +1,1144 @@
+module Marlowe.Blockly where
+
+import Prelude
+
+import Blockly (AlignDirection(..), Arg(..), BlockDefinition(..), block, blockType, category, colour, defaultBlockDefinition, getBlockById, initializeWorkspace, name, render, style, x, xml, y)
+import Blockly.Generator (Generator, Input, clearWorkspace, compose2, connectToOutput, connectToPrevious, fieldName, fieldRow, getFieldValue, getInputWithName, inputList, inputName, insertGeneratorFunction, mkGenerator, setFieldText, statementToCode)
+import Blockly.Types (Block, BlocklyState, Workspace)
+import Control.Alternative ((<|>))
+import Data.Array as Array
+import Data.Bifunctor (lmap, rmap)
+import Data.Either (Either)
+import Data.Maybe (Maybe(..))
+import Data.Newtype (unwrap)
+import Effect (Effect)
+import Halogen.HTML (HTML)
+import Halogen.HTML.Properties (id_)
+import Marlowe.Parser as Parser
+import Marlowe.Types (Contract(..), IdChoice(..), Observation(..), Value(..))
+import Record (merge)
+import Text.Parsing.Parser (Parser, runParser)
+import Text.Parsing.Parser.Basic (parens)
+
+data BlockType
```

Comment:
I'm thinking you should split this into:

```
data BlockType
  = ContractBlock ContractType
  | ObservationBlock ObservationType
  | ValueBlock ValueType

...
```

If you did that, you could generate `toolbox` automatically from `blockDefinitions`, for starters.

Reply:
done

Reply:
Schweet.